lib/simulator-xcode-6.js

```diff
@@ -184,18 +187,33 @@ class SimulatorXcode6 {
     return _.compact(existence).length !== files.length;
   }
 
-  //
-  // setPreferences () {
-  //
-  // }
-  //
-  // setLocationPreferences () {
-  //
-  // }
-  //
-  // setSafariPreferences () {
-  //
-  // }
+  async run () {
+    const STARTUP_TIMEOUT = 3000;
+    const BOOTING_TIMOUT = 20000;
+    const BOOTED_POLLING_INTERVAL = 200;
+
+    // start simulator
+    let simulatorApp = path.resolve(await getXcodePath(), 'Applications', 'Simulator.app');
+    let args = [simulatorApp, '--args', '-CurrentDeviceUDID', this.udid];
+    log.info(`starting simulator with command: open ${args.join(' ')}`);
+    await exec('open', args, {timeout: STARTUP_TIMEOUT});
+
+    //wait for it to be 'booted'
+    let checkBooted = async function () {
+      let stat = await this.stat();
+      return new B((resolve, reject) => {
+        if (stat.state === 'Booted') {
+          resolve();
+        } else {
+          reject();
+        }
+      });
+    }.bind(this);
+
+    await retryInterval(parseInt(BOOTING_TIMOUT/BOOTED_POLLING_INTERVAL), BOOTED_POLLING_INTERVAL, () => {
```

Comment:
i think by our conventions this should be an async function and `return await checkBooted();`